src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala

```diff
@@ -470,12 +478,18 @@ object SparkExpressionToCExpression {
       case BooleanType   => VeScalarType.veNullableInt
       case StringType    => VeString
       case TimestampType => VeScalarType.veNullableLong
+      case _ => sys.error(s"SparkExpressionToCExpression.sparkTypeToVeType: " +
+        "Could not match ${dataType}, " +
+        "class ${dataType.getClass}")
     }
   }
   def sparkSortDirectionToSortOrdering(sortDirection: SortDirection): SortOrdering = {
     sortDirection match {
       case expressions.Ascending  => Ascending
       case expressions.Descending => Descending
+      case _ => sys.error(s"SparkExpressionToCExpression.sparkSortDirectionToSortOrdering: " +
```

Comment:
This case is covered by Scala's pattern matcher

Reply:
I see your point.